### DISCUSSION TITLE
Are we able to dedicate a thread to a specific core?

### DISCUSSION THREAD

**Discussion by jimkoke** [micropython/micropython#15387] [Libraries & Drivers]
My question is simple - I have a thread, I want to tell micropython to run that thread on a single core and only that thread on that core, so that, with proper resource management on my part, that thread runs without having to give up the core?

[Edit] This will be running on a Raspberry Pi Pico.

I've read dozens of articles on this subject (as close I can get to it in searches, anyway), humorlessly reading multiple "original" articles that copied the same source code down to the spelling errors and providing keyhole commentary. These article state they are about using both cores, however, the code does not reflect that, only demonstrating how to start a second thread. There is nothing, that I could see, that locks the second thread to a specific core, and so I don't know if it is possible. These articles tend to be focused on multi-threading, but I want to know if the scheduler can lock code to cores. I've done this on larger systems using "processor sets" to manage simulation processes across 8 cores.

**Comment by Josverl** [comment]
>  I have a thread, I want to tell micropython to run that thread on a single core and only that thread on that core

Which micropython port(s) are you considering? This is relevant as different ports handle CPU cores different

**Reply by jimkoke** [reply]
Doh! Sorry - I'm looking at running on a Raspberry Pi Pico W

**Comment by GitHubsSilverBullet** [comment]
> My question is simple - I have a thread, I want to tell micropython to run that thread on a single core and only that thread on that core, so that, with proper resource management on my part, that thread runs without having to give up the core?

At least on a RP2040 processor, everything is running on core 0.
Core 1 is dead like a dodo. If one starts a job using `_thread.start_new_thread()` then that one runs on core 1 and only on core 1. If you're concerned about ISRs and where those execute, that's a bit muddy water. One would have to write a sufficiently good test program.

**Comment by Tangerino** [comment]
What about ESP32?

**Comment by bixb922** [comment]
Esp32 can run Micropython only on one core. The second core is reserved for operating system tasks.
